ksql-engine/src/main/java/io/confluent/ksql/function/UdfLoader.java

```diff
@@ -266,9 +267,12 @@ private void addFunction(final UdfDescription classLevelAnnotaion,
         parameters,
         classLevelAnnotaion.name(),
         udfClass,
-        () -> {
-          final PluggableUdf theUdf
-              = new PluggableUdf(udf, instantiateUdfClass(method, classLevelAnnotaion));
+        ksqlConfig -> {
+          final Object actualUdf = instantiateUdfClass(method, classLevelAnnotaion);
+          if (actualUdf instanceof Configurable) {
+            ((Configurable)actualUdf).configure(ksqlConfig.originals());
```

Comment:
Is it safe to send all of the config to the user's UDF? We may have sensitive info such as ACL secret.
Maybe you can have another method that only passed the config properties that are ok to pass.

Reply:
This is an interesting one. Not sure what is the best approach. It might be that the UDF needs security information to do its job. (Think UDF that calls out to external service, e.g. DB, or maybe Kafka).

Humm....

How about we limit passed config to anything prefixed with `ksql.functions`?